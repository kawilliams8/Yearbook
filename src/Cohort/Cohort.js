import React from 'react';
import Person from '../Person/Person';
import './Cohort.css';

const Cohort = (props) => {

  let cohortCards = props.people.map(person => {
    let photo;
    props.position === 'staff' ? 
      photo = person.photo :
      photo = `https://picsum.photos/id/${person.id}/200/300`;
    return <Person 
      key={person.id} 
      id={person.id} 
      name={person.name} 
      quote={person.quote} 
      superlative={person.superlative} 
      photo={photo} 
      removeStudent={props.removeStudent}
      position={props.position}/>
  })
    return (
      <div className="Cohort">
        {cohortCards}
      </div>
    )
}

export default Cohort;
