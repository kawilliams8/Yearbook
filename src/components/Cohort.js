import React from 'react';
import Person from './Person';
import './Cohort.css';

const Cohort = (props) => {
  let cohortCards = props.people.map(person => {
    let photo;
    if (props.position === 'staff') {
      photo = person.photo;
    } else {
      photo = `https://picsum.photos/id/${person.id}/200/300`;
    }
  
    let {id, name, quote, superlative} = person;
    return <Person key={id} name={name} quote={quote} superlative={superlative} photo={photo}/>
  })

  return (
    <div className="Cohort">
      {cohortCards}
    </div>
  )
}

export default Cohort;
