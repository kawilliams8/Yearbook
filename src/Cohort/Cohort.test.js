import React from 'react';
import { shallow } from 'enzyme';
import Cohort from './Cohort';

describe('Cohort', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const students = [
        { id: 21, name: 'Abigail', quote: 'Howdy.', superlative: 'Most Likely to', photo: 'https://placekitten.com/200/300' },
        { id: 22, name: 'Brian', quote: 'Hello.', superlative: 'Most Likely to', photo: 'https://placekitten.com/200/300' },
        { id: 23, name: 'Charlotte', quote: 'Hey.', superlative: 'Most Likely to', photo: 'https://placekitten.com/200/300' },
      ]

    const wrapper = shallow(<Cohort 
      position="students"
      people={students}
      removeStudent={jest.fn()}
      />);

    expect(wrapper).toMatchSnapshot();
  });
});