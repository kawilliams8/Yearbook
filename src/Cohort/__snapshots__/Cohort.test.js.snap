// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should match the snapshot with all data passed in correctly 1`] = `
<div
  className="Cohort"
>
  <Person
    id={21}
    key="21"
    name="Abigail"
    photo="https://picsum.photos/id/21/200/300"
    position="students"
    quote="Howdy."
    removeStudent={[MockFunction]}
    superlative="Most Likely to"
  />
  <Person
    id={22}
    key="22"
    name="Brian"
    photo="https://picsum.photos/id/22/200/300"
    position="students"
    quote="Hello."
    removeStudent={[MockFunction]}
    superlative="Most Likely to"
  />
  <Person
    id={23}
    key="23"
    name="Charlotte"
    photo="https://picsum.photos/id/23/200/300"
    position="students"
    quote="Hey."
    removeStudent={[MockFunction]}
    superlative="Most Likely to"
  />
</div>
`;
